### PR TITLE
upgrade to Node 0.12

### LINKED ISSUE #208: Markdown parser needs replacement
Apparently robotskirt is abandoned in favour of https://github.com/hoedown/node-hoedown. Current setup works only with Node `0.10`.

I am [looking for a alternative](http://softwarerecs.stackexchange.com/q/20702/27)... let me know if some of you have another suggestion.

So far could be https://github.com/chjj/marked, or https://github.com/hoedown/node-hoedown

 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/23200716-markdown-parser-needs-replacement?utm_campaign=plugin&utm_content=tracker%2F130405&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F130405&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SergioCrisostomo**
Replaced incompatible dependencies:

 - robotskirt
 - sync-prompt

Working on Node.js `0.12.7` and io.js `2.3.3`.

(fixes #208)

### CODE REVIEW

File: build/api_keys.js [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
 
-function checkKeys(api, name){
-	Object.keys(api).forEach(function(key){
-		api[key] = prompt('Please insert ' + name + '\'s ' + key + ': ') || api[key];
-	});
+function checkKeys(site, done){
+    var keys = Object.keys(APIkeys[site]);
+    (function question() {
+        var key = keys.shift();
+		if (!key) return done();
```
Comment (arian) [inline build/api_keys.js]: Indentation

File: build/api_keys.js [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
+	    input: process.stdin,
+	    output: process.stdout
+	});
+
+	(function iterator(){
+	    var service = services.shift();
+	    if (service) return question(service, iterator);
+
+	    rli.close();
+		fs.writeFileSync(APIpath, JSON.stringify(APIkeys, null, 4));
```
Comment (arian) [inline build/api_keys.js]: indentation

**Comment by arian** [comment]
If all the markdown output is exactly the same, this seems ok!

**Comment by SergioCrisostomo** [comment]
Thank you for feedback @arian!

Fixed indentation. The markdown output looks consistent. 
Did some diff-checks with the HTML from both and looks good.

**Pull request merged**